spec/acceptance/locales_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'spec_helper'
+require 'spec_helper_acceptance'
+require 'beaker/i18n_helper'
+require 'pry'
+
+describe 'i18n Testing', if: (fact('osfamily') == 'Debian' || fact('osfamily') == 'RedHat') && (Gem::Version.new(puppet_version) >= Gem::Version.new('4.10.5')) do
+  before :all do
+    hosts.each do |host|
+      on(host, "sed -i \"96i FastGettext.locale='ja'\" /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet.rb")
+      change_locale_on(host, 'ja_JP.utf-8')
+    end
+  end
+
+  shared_examples 'has_error' do |path, pp, error|
+    before :all do
+      shell("rm #{path}", acceptable_exit_codes: [0, 1, 2]) if Dir.exist?(path)
+    end
+    after :all do
+      shell("cat #{path}", acceptable_exit_codes: [0, 1, 2])
+      shell("rm #{path}", acceptable_exit_codes: [0, 1, 2])
+    end
+
+    it 'applies the manifest and gets a failure message' do
+      expect(apply_manifest(pp, expect_failures: true).stderr).to match(error)
+    end
+  end
+
+  context 'display Japanese error' do
```

Comment:
Your test's need to check two separate messages. 
Despite the different manifest's they are both technically testing the same thing.